### PR TITLE
Oceanwater 777 refactor action code

### PR DESCRIPTION
**Pull request by anjan1984**
## Linked Issues:
| EPIC ⚡ | [OCEANWATER-XXX](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-XXX) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-777](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-777) |
| Github :octocat:  | # |

## Summary of Changes
* Simplifying the action code. Combined all action servers into one file and combined all trajectories generated into one file. The clients are still in separate files for ease of use.

## Test
* All ros action tests
* roslauch ow europa_terminator_workspace.launch
* In a separate terminal:

cd PATH_TO_WORKSPACE/oceanwaters_ws
source devel/setup.bash

Go to the scripts folder of the ow_lander.

cd src/ow_simulator/ow_lander/scripts

- python unstow_action_client.py
- python guarded_move_action_client.py
- python grind_action_client.py
- python dig_circular_action_client.py
- python dig_linear_action_client.py
- python deliver_action_client.py
- python stow_action_client.py

### CODE REVIEW

**Comment by kmdalal** [comment]
@anjan1984 Do you think this needs three reviewers?  Two is standard for moderate changes, 3+ for big changes.  I can test this but only give the code a light review as I'm not much of a Python programmer.

**Comment by anjan1984** [comment]
> @anjan1984 Do you think this needs three reviewers? Two is standard for moderate changes, 3+ for big changes. I can test this but only give the code a light review as I'm not much of a Python programmer.

No ... 3 reviewers are not necessary for this

File: ow_lander/launch/spawn.launch [hunk 1]
```diff
@@ -102,12 +102,6 @@
...
-  <node pkg="ow_lander" name="Stow" type="stow_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="Deliver" type="deliver_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="DigLinear" type="dig_linear_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="DigCircular" type="dig_circular_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="Grind" type="grind_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="GuardedMove" type="guarded_move_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
-  <node pkg="ow_lander" name="AntennaPanTiltAction" type="antenna_pan_tilt_action_server.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
+  <node pkg="ow_lander" name="AllArmActionServers" type="all_arm_action_servers.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
```
Comment (Samahu) [inline ow_lander/launch/spawn.launch:105]: I would drop the "all_" from the `all_arm_action_servers.py`
```suggestion
  <node pkg="ow_lander" name="ArmActionServers" type="arm_action_servers.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
```

File: ow_lander/scripts/all_action_trajectories.py [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+from geometry_msgs.msg import Quaternion
+from shape_msgs.msg import SolidPrimitive
+from moveit_msgs.msg import RobotTrajectory
+from trajectory_msgs.msg import JointTrajectory
+from trajectory_msgs.msg import JointTrajectoryPoint
+from std_msgs.msg import Header
+
+
+
+deliver_sample_traj = RobotTrajectory()
```
Comment (Samahu) [inline ow_lander/scripts/all_action_trajectories.py:24]: This seems unreferenced to me?
```suggestion
```

File: ow_lander/scripts/all_action_trajectories.py [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+    # Rotate dist to dig
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    dist_now = start_state[3]
+    plan_f = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, dist_now + 2*math.pi/3)
+    circ_traj = cascade_plans (circ_traj, plan_f)
+
+  return circ_traj
+
+
+dig_linear_traj = RobotTrajectory()
```
Comment (Samahu) [inline ow_lander/scripts/all_action_trajectories.py:304]: This also seem unreferenced?
```suggestion
```

File: ow_lander/scripts/all_action_trajectories.py [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+  #  rotate to dig out
+  cs, start_state, current_pose = calculate_joint_state_end_pose_from_plan_arm (robot, dig_linear_traj, move_arm, moveit_fk)
+  plan_g = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, math.pi/2)
+  dig_linear_traj = cascade_plans (dig_linear_traj, plan_g)
+  
+  return dig_linear_traj
+  
+  
+  
+grind_traj = RobotTrajectory()
```
Comment (Samahu) [inline ow_lander/scripts/all_action_trajectories.py:548]: This one too
```suggestion
```

File: ow_lander/scripts/all_action_trajectories.py [hunk 5]
```diff
@@ -0,0 +1,853 @@
...
+  plan_d = go_to_Z_coordinate(move_grinder, cs, joint_goal, x_start, y_start, 0.22, False)
+  grind_traj = cascade_plans (grind_traj , plan_d)
+
+  return grind_traj
+
+
+
+
+pre_guarded_move_traj = RobotTrajectory()
+guarded_move_traj = RobotTrajectory()
```
Comment (Samahu) [inline ow_lander/scripts/all_action_trajectories.py:689]: Unreferenced?
```suggestion
```

File: ow_lander/scripts/all_arm_action_servers.py [hunk 6]
```diff
@@ -0,0 +1,667 @@
+#!/usr/bin/env python2
+
+# The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+# Research and Simulation can be found in README.md in the root directory of
+# this repository.
+
+import rospy
+import actionlib
+from ow_lander.msg import *
+import time
```
Comment (Samahu) [inline ow_lander/scripts/all_arm_action_servers.py:10]: Remove and replace any references to **time** with `rospy.get_time()` and `rospy.sleep()`
```suggestion
```

**Review by Samahu — CHANGES_REQUESTED** [review]
minor improvements requested

File: ow_lander/launch/spawn.launch (hunk 1 above)
Comment (anjan1984) [reply ow_lander/launch/spawn.launch]: Done!

File: ow_lander/scripts/all_action_trajectories.py (hunk 2 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:24]: Done!

File: ow_lander/scripts/all_action_trajectories.py (hunk 3 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:304]: Done!

File: ow_lander/scripts/all_action_trajectories.py (hunk 4 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:548]: removed!

File: ow_lander/scripts/all_action_trajectories.py (hunk 5 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:689]: removed!

File: ow_lander/scripts/all_arm_action_servers.py (hunk 6 above)
Comment (anjan1984) [reply ow_lander/scripts/all_arm_action_servers.py]: Done!

File: ow_lander/scripts/all_action_trajectories.py [hunk 7]
```diff
@@ -0,0 +1,853 @@
...
+  if (is_shou_yaw_goal_in_range(joint_goal) == False):
+    return False
+
+  joint_goal[constants.J_SCOOP_YAW] = 0
+  
+  plan = move_arm.plan(joint_goal)
+  return plan
+
+
+def dig_circular(move_arm, move_limbs, robot, moveit_fk, args):
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:210]: This function is quite long.  Perhaps break it into two, for parallel and non-parallel?

File: ow_lander/scripts/all_action_trajectories.py [hunk 8]
```diff
@@ -0,0 +1,853 @@
...
+def calculate_joint_state_end_pose_from_plan_arm (robot, plan, move_arm, moveit_fk):
+  ''' 
+  calculate the end pose (position and orientation), joint states and robot states
+  from the current plan
+  inputs:  current plan, robot, arm interface, and moveit forward kinematics object
+  outputs: goal_pose, robot state and joint states at end of the plan
+  '''  
+  #joint_names: [j_shou_yaw, j_shou_pitch, j_prox_pitch, j_dist_pitch, j_hand_yaw, j_scoop_yaw]
+  #robot full state name: [j_ant_pan, j_ant_tilt, j_shou_yaw, j_shou_pitch, j_prox_pitch, j_dist_pitch, j_hand_yaw,
+  #j_grinder, j_scoop_yaw]
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:35]: Is this comment useful and does it need to be repeated in many functions?

File: ow_lander/scripts/all_action_trajectories.py [hunk 9]
```diff
@@ -0,0 +1,853 @@
...
+    plan_f = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, dist_now + 2*math.pi/3)
+    circ_traj = cascade_plans (circ_traj, plan_f)
+
+  return circ_traj
+
+
+dig_linear_traj = RobotTrajectory()
+
+
+def calculate_joint_state_end_pose_from_plan_arm (robot, plan, move_arm, moveit_fk):
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:307]: Every line that calls this function is quite long.  Suggest shortening the name, e.g. `joint_state_end_pose` or even `end_pose` if the rest is inferable by context.

File: ow_lander/scripts/all_action_trajectories.py [hunk 10]
```diff
@@ -0,0 +1,853 @@
...
+  
+  plan_d = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, -math.pi/2)
+
+  dig_linear_traj = cascade_plans (dig_linear_traj, plan_d)
+  
+  ##########################Once aligned to trench goal, 
+  ##########################place hand above the desired start point
+  alpha = math.atan2(constants.WRIST_SCOOP_PARAL, constants.WRIST_SCOOP_PERP)
+  distance_from_ground = constants.ROT_RADIUS * \
+      (math.cos(alpha) - math.sin(alpha))
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:507]: Indentation off.  Could be added to previous line.

File: ow_lander/scripts/all_action_trajectories.py [hunk 11]
```diff
@@ -0,0 +1,853 @@
...
+  goal_pose.position.z -= direction_z*search_distance
+
+  move_arm.set_pose_target(goal_pose)
+  plan_c  = move_arm.plan()
+  
+  guarded_move_traj = cascade_plans (pre_guarded_move_traj, plan_c)
+
+  #n_points = len(pre_guarded_move_traj.joint_trajectory.points)
+  pre_guarded_move_end_time = pre_guarded_move_traj.joint_trajectory.points[len(pre_guarded_move_traj.joint_trajectory.points)-1].time_from_start
+  guarded_move_end_time = guarded_move_traj.joint_trajectory.points[len(guarded_move_traj.joint_trajectory.points)-1].time_from_start
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:772]: Lines too long.

File: ow_lander/scripts/all_action_trajectories.py [hunk 12]
```diff
@@ -0,0 +1,853 @@
+ #!/usr/bin/env python2
```
Comment (kmdalal) [inline ow_lander/scripts/all_action_trajectories.py:1]: Minor suggestion: rename file to just `action_trajectories.py`.

File: ow_lander/scripts/arm_action_servers.py [hunk 13]
```diff
@@ -0,0 +1,666 @@
...
+        plan = self._update_motion()
+        if plan is None: 
+            self._server.set_aborted(self._result)
+            return 
+        success = False
+
+        self.trajectory_async_executer.execute(plan.joint_trajectory,
+                                           done_cb=None,
+                                           active_cb=None,
+                                           feedback_cb=self.trajectory_async_executer.stop_arm_if_fault)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:75]: Indentation off.

File: ow_lander/scripts/arm_action_servers.py [hunk 14]
```diff
@@ -0,0 +1,666 @@
...
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:152]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 15]
```diff
@@ -0,0 +1,666 @@
...
+            return 
+        success = False
+        switch_success = self.switch_controllers('grinder_controller', 'arm_controller')
+        if not switch_success:
+          return False, "Failed switching controllers"
+      
+        self.trajectory_async_executer.execute(self.current_traj.joint_trajectory,
+                                           done_cb=None,
+                                           active_cb=None,
+                                           feedback_cb=self.trajectory_async_executer.stop_arm_if_fault)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:249]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 16]
```diff
@@ -0,0 +1,666 @@
...
+
+
+class GuardedMoveActionServer(object):
+    
+    def __init__(self,name):
+        self._action_name = name
+        self._server = actionlib.SimpleActionServer(self._action_name, 
+                                     ow_lander.msg.GuardedMoveAction, 
+                                     execute_cb=self.on_guarded_move_action, 
+                                     auto_start = False)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:287]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 17]
```diff
@@ -0,0 +1,666 @@
...
+        self._estimated_plan_fraction_completed = 0.0
+        #ratio between guarded pre-guarded move trajectory and the whole trajectory
+        self._guarded_move_plan_ratio = 0.0  
+        self.trajectory_async_executer = TrajectoryAsyncExecuter()
+        self.trajectory_async_executer.connect("arm_controller")
+        self.guarded_move_traj = RobotTrajectory()
+        self.ground_detector = GroundDetector()
+        self.pos = Point()
+        self.guarded_move_pub = rospy.Publisher(
+        '/guarded_move_result', GuardedMoveFinalResult, queue_size=10)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:304]: Would read better as one line, or split differently.

File: ow_lander/scripts/arm_action_servers.py [hunk 18]
```diff
@@ -0,0 +1,666 @@
...
+        success = False
+
+
+        # detection
+        self.ground_detector.reset()
+
+        self.trajectory_async_executer.execute(self.guarded_move_traj.joint_trajectory,
+                                           done_cb=self.handle_guarded_move_done,
+                                           active_cb=None,
+                                           feedback_cb=self.handle_guarded_move_feedback)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:373]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 19]
```diff
@@ -0,0 +1,666 @@
...
+        plan = self._update_motion(goal)
+        if self.current_traj == False: 
+            self._server.set_aborted(self._result)
+            return 
+        success = False
+
+        self.trajectory_async_executer.execute(self.current_traj.joint_trajectory,
+                                           done_cb=None,
+                                           active_cb=None,
+                                           feedback_cb=self.trajectory_async_executer.stop_arm_if_fault)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:472]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 20]
```diff
@@ -0,0 +1,666 @@
...
+        self._server.publish_feedback(self._fdbk)
+
+        
+        
+    def _update_motion(self, goal):
+        
+        print("DigLinear activity started")
+        self.current_traj  = all_action_trajectories.dig_linear(self._interface.move_arm,
+                                                          self._interface.robot,  
+                                                          self._interface.moveit_fk, goal)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:533]: Indent.

File: ow_lander/scripts/arm_action_servers.py [hunk 21]
```diff
@@ -0,0 +1,666 @@
...
+        if self.current_traj == False: 
+            return 
+        else:
+            n_points = len(self.current_traj.joint_trajectory.points)
+            start_time =   self.current_traj.joint_trajectory.points[0].time_from_start
+            end_time =      self.current_traj.joint_trajectory.points[n_points-1].time_from_start
+            self._timeout =  (end_time -start_time)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:540]: The spacing/alignment could be more consistent here, and in many other similar places.

File: ow_lander/scripts/arm_action_servers.py [hunk 22]
```diff
@@ -0,0 +1,666 @@
...
+            self._server.set_succeeded(self._result)
+        else:
+            rospy.loginfo('%s: Failed' % self._action_name)
+            self._server.set_aborted(self._result)      
+
+class DeliverActionServer(object):
+    
+    def __init__(self,name):
+        self._action_name = name
+        self._server = actionlib.SimpleActionServer(self._action_name, ow_lander.msg.DeliverAction, execute_cb=self.on_deliver_action, auto_start = False)
```
Comment (kmdalal) [inline ow_lander/scripts/arm_action_servers.py:584]: Line too long.

**Review by kmdalal — APPROVED** [review]
All the prescribed tests worked, as well as two substantial PLEXIL plans.   Just minor code comments.   Nice work!

**Comment by kmdalal** [comment]
One more code formatting suggestion: be consistent about the number of blank lines between functions, etc.

**Review by Samahu — APPROVED** [review]

File: ow_lander/scripts/arm_action_servers.py [hunk 23]
```diff
@@ -0,0 +1,666 @@
...
+        self._ls =  self._current_link_state._link_value
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+        
+    def _update_motion(self):
+
+        print("Unstow arm activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:52]: Would rather see rospy logging used for all print functions.
```suggestion
        rospy.loginfo("Unstow arm activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 24]
```diff
@@ -0,0 +1,666 @@
...
+ 
+        self._ls =  self._current_link_state._link_value
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+    def _update_motion(self):
+
+        print("Stow arm activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:127]:
```suggestion
        rospy.loginfo("Stow arm activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 25]
```diff
@@ -0,0 +1,666 @@
...
+    def switch_controllers(self, start_controller, stop_controller):
+        rospy.wait_for_service('/controller_manager/switch_controller')
+        success = False
+        try:
+          switch_controller = rospy.ServiceProxy(
+          '/controller_manager/switch_controller', SwitchController)
+          success = switch_controller(
+          [start_controller], [stop_controller], 2, False, 1.0)
+        except rospy.ServiceException as e:
+          print("switch_controllers error: %s" % e)
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:213]:
```suggestion
          rospy.logerror("switch_controllers error: %s", e)
```

File: ow_lander/scripts/arm_action_servers.py [hunk 26]
```diff
@@ -0,0 +1,666 @@
...
+        finally:
+          # This sleep is a workaround for "start point deviates from current robot
+          #  state" error on dig_circular trajectory execution.
+          rospy.sleep(0.2)
+          return success
+        
+        
+    def _update_motion(self, goal):
+        print("Grind activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:222]:
```suggestion
        rospy.loginfo("Grind activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 27]
```diff
@@ -0,0 +1,666 @@
...
+        self._ls =  self._current_link_state._link_value
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+        
+        
+    def _update_motion(self, goal):
+        print("Guarded move activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:343]:
```suggestion
        rospy.loginfo("Guarded move activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 28]
```diff
@@ -0,0 +1,666 @@
...
+    def switch_controllers(self, start_controller, stop_controller):
+        rospy.wait_for_service('/controller_manager/switch_controller')
+        success = False
+        try:
+          switch_controller = rospy.ServiceProxy(
+          '/controller_manager/switch_controller', SwitchController)
+          success = switch_controller(
+          [start_controller], [stop_controller], 2, False, 1.0)
+        except rospy.ServiceException as e:
+          print("switch_controllers error: %s" % e)
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:428]:
```suggestion
          rospy.logerror("switch_controllers error: %s", e)
```

File: ow_lander/scripts/arm_action_servers.py [hunk 29]
```diff
@@ -0,0 +1,666 @@
...
+        self._ls =  self._current_link_state._link_value
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+        
+        
+    def _update_motion(self, goal):
+        print("DigCircular activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:447]:
```suggestion
        rospy.loginfo("DigCircular activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 30]
```diff
@@ -0,0 +1,666 @@
...
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+        
+        
+    def _update_motion(self, goal):
+        
+        print("DigLinear activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:530]:
```suggestion
        rospy.loginfo("DigLinear activity started")
```

File: ow_lander/scripts/arm_action_servers.py [hunk 31]
```diff
@@ -0,0 +1,666 @@
...
+        self._ls =  self._current_link_state._link_value
+        self._fdbk.current.x = self._ls.x
+        self._fdbk.current.y = self._ls.y
+        self._fdbk.current.z = self._ls.z
+        self._server.publish_feedback(self._fdbk)
+
+        
+        
+    def _update_motion(self, goal):
+        print("Deliver sample activity started")
```
Comment (AstroStucky) [inline ow_lander/scripts/arm_action_servers.py:607]:
```suggestion
        rospy.loginfo("Deliver sample activity started")
```

**Review by AstroStucky — APPROVED** [review]
I kept my review brief, and was mostly checking that OCEANWATER-803 was also addressed by this refactor. I only reviewed code, did not test. 

Please address the code comments I made on the use of print functions.

Either in this PR or a later PR, I think arm_action_servers.py could be refactored with an ArmActionServerBase class that all arm actions inherit from. I see a lot of code repeated for each class, and implementing a base class should reduce the repetition.

File: ow_lander/scripts/all_action_trajectories.py (hunk 7 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:210]: parallel is an argument .. it is same implementation as services.... so making this change.

File: ow_lander/scripts/all_action_trajectories.py (hunk 8 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:35]: I think it is useful... there is a difference between the joints and the full robot state.. and the order of the names differ based on the control group chosen..... thus it is a good idea to state the joint names.....

File: ow_lander/scripts/all_action_trajectories.py (hunk 9 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:307]: the long name specifies what the function is doing....

File: ow_lander/scripts/all_action_trajectories.py (hunk 10 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:507]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 13 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:75]: fixed

File: ow_lander/scripts/arm_action_servers.py (hunk 14 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:152]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 15 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:249]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 16 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:287]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 17 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:304]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 18 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:373]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 19 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:472]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 20 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:533]: downloaded python suggested app to auto -intend all in vs code... should fix all

File: ow_lander/scripts/arm_action_servers.py (hunk 21 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:540]: fixed

File: ow_lander/scripts/arm_action_servers.py (hunk 22 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:584]: done!

File: ow_lander/scripts/all_action_trajectories.py (hunk 11 above)
Comment (anjan1984) [reply ow_lander/scripts/all_action_trajectories.py:772]: > One more code formatting suggestion: be consistent about the number of blank lines between functions, etc.

done..

File: ow_lander/scripts/arm_action_servers.py (hunk 28 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:428]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 29 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:447]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 30 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:530]: done!

File: ow_lander/scripts/arm_action_servers.py (hunk 31 above)
Comment (anjan1984) [reply ow_lander/scripts/arm_action_servers.py:607]: done!

**Pull request merged**